content/augmentor.js

```diff
@@ -0,0 +1,336 @@
+        return Runner.execute(FETCH_IMAGE, badge.img)
+          .then((src) => ({ ...badge, src }));
+      });
+
+    const tokensPromises = tokens
+      .map((token) => {
+        return Runner.execute(FETCH_IMAGE, token.img)
+          .then((src) => ({ ...token, src }));
+      });
+
+    const { height = 16 } = node.getBoundingClientRect();
+
+    const icon = blockies({
+      seed: (address || '').toLowerCase(),
+      size: 8,
+      scale: 8
+    }).toDataURL();
+
+    Promise
+      .all([ Promise.all(badgesPromises), Promise.all(tokensPromises) ])
+      .then(([ badgesData, tokensData ]) => {
+        const badgesElements = badgesData.map((badge) => Augmentor.getBadge(badge, height));
+
+        // The Ethereum Addres Identity Icon
+        const blockieElement = Augmentor.getBadge({ src: icon, title: address }, height);
+
+        // The Badges container
+        const badgesElement = document.createElement('span');
+        badgesElement.className = styles.badges;
+        badgesElements.forEach((elt) => badgesElement.appendChild(elt));
+
+        // The Account Card
+        const cardElement = Augmentor.getAccountCard(resolved[key], icon, badgesData, tokensData);
+
+        // The main Container
+        const iconsElement = document.createElement('span');
+        iconsElement.setAttribute('data-parity-touched', true);
+        iconsElement.className = styles.icons;
+        iconsElement.appendChild(blockieElement);
+        iconsElement.appendChild(badgesElement);
+        iconsElement.appendChild(cardElement);
+
+        iconsElement.onclick = (event) => {
+          event.preventDefault();
+          event.stopPropagation();
+
+          const { currentTarget } = event;
+
+          const selectedText = window.getSelection().toString();
+
+          // Don't close if text is selected
+          if (selectedText) {
+            return false;
+          }
+
+          const classes = currentTarget.className.split(' ').map((className) => className.trim());
+
+          if (classes.includes(styles.expanded)) {
+            currentTarget.className = classes.filter((className) => className !== styles.expanded).join(' ');
+          } else {
+            currentTarget.className = classes.concat(styles.expanded).join(' ');
+          }
+        };
+
+        const container = document.createElement('span');
+        container.className = styles.container;
+
+        container.appendChild(node.cloneNode(true));
+        container.appendChild(iconsElement);
+
+        node.parentElement.insertBefore(container, node);
+        node.parentElement.removeChild(node);
+
+        Augmentor.positionNode(badgesElement, iconsElement, { scale: 1.5, forceBottom: true });
+        Augmentor.positionNode(cardElement, iconsElement, { scale: 4 });
+      });
+  }
+
+  static positionNode (node, container, options = {}) {
+    const { scale = 1.5, offset = 5, forceBottom = false } = options;
+
+    const nodeRect = node.getBoundingClientRect();
```

Comment:
This might break if augmented node is not visible, we should probably do positioning on `mouseenter`.

We may also save a lot of DOM nodes if we build those elements on `mouseenter` instead too.